resources/schema.py

```diff
@@ -328,6 +340,52 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         return DeleteBerthTypeMutation()
 
 
+class CreateHarborMutation(graphene.ClientIDMutation):
+    class Input(AbstractAreaInput):
+        municipality = graphene.String()
+        image_file = graphene.String()
+        availability_level = graphene.ID()
+        number_of_places = graphene.Int()
+        maximum_width = graphene.Int()
+        maximum_length = graphene.Int()
+        maximum_depth = graphene.Int()
+        name = graphene.String()
+        street_address = graphene.String()
+
+    harbor = graphene.Field(HarborNode)
+
+    @classmethod
+    @login_required
+    @superuser_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO: Should check if the user has permissions to
+        # delete the specific resource
+        lang = get_language()
+
+        availability_level_id = kwargs.pop("availability_level", None)
+        availability_level = None
+        municipality_id = kwargs.pop("municipality", None)
+
+        if availability_level_id:
+            try:
+                availability_level = AvailabilityLevel.objects.language(lang).get(
+                    pk=availability_level_id
+                )
+            except AvailabilityLevel.DoesNotExist as e:
+                raise VenepaikkaGraphQLError(e)
+
+        params = {**kwargs}
```

Comment:
why do you need to init a new dict? `kwargs` is dict already and if you `pop`ed the right things it should work ok 🙂

Reply:
True 🤔 I was thinking to do it like that to keep the `kwargs` immutable, but if I'm popping then it doesn't matter that much.